### PR TITLE
Make the shell be dynamic to window size

### PR DESCRIPTION
**Pull request by Sir-Thom**
I added the function `async_create_shell` which create the shell and allow it to resize and size itself  correctly. Also  if the user have set `neofetch` in is  `.bashrc`  it will be size correctly at the launch of the terminal. Also I added  to the shell the  `TERM`  env so if you build the program,  you can use `cls` or `clear.` Finally I have updated tauri and other packages.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -11,34 +11,38 @@ const term = new Terminal({
...
 
 // Make the terminal fit all the window size
-function fitTerminal(){
+async function fitTerminal() {
   fitAddon.fit();
   void invoke<string>("async_resize_pty", {
     rows: term.rows,
     cols: term.cols,
   });
+  // Create a shell if it doesn't exist
```
Comment (marc2332) [inline src/main.ts]: What's the point of this? I don't get it

**Review by marc2332 — CHANGES_REQUESTED** [review]
I don't understand what's the goal of this PR, you are creating a shell everytime you resize the Window

File: src/main.ts (hunk 1 above)
Comment (Sir-Thom) [reply src/main.ts]: Sorry that was a mistake on my part. Now, the shell should be created before the  the writeToPty.

**Review by marc2332 — CHANGES_REQUESTED** [review]
I still don't see why is needed to create a new command `async_create_shell`, what's wrong with creating it in the main function

**Comment by Sir-Thom** [comment]
> I still don't see why is needed to create a new command `async_create_shell`, what's wrong with creating it in the main function

There was a bug causing every program/code writing text of the screen not be completely written. For example if you have `noefetch` in the `.bashrc` only some part of it would be written when the tauri Windows is created. I have some pictures to show what i mean i will upload later and , the solution I came up to solve it was to create a function for the shell. Finally, I add $TERM env variable so when you build it you  can use some shortcut like clear to clean the window.

@marc2332 here the of what the bug was doing

**Comment by marc2332** [comment]
Ahh right, it makes sense to create the shell after the emulator has been created, otherwise it will not print the initial shell  output, just as you described

I'll review this asap, I think it's a good idea to create the shell from a command instead of main

Nice work :muscle:

**Review by marc2332 — APPROVED** [review]

**Pull request merged**